### ISSUE TITLE
Exponentially slow type evaluation with non-total TypedDict

### ISSUE THREAD

**Issue by rslinckx** [microsoft/pyright#3734]
Here's a minimal reproduction (the code is pointless but is similar in essence to something we have in the real codebase):

```
from typing import TypedDict


class GroupsSettingsDict(TypedDict, total=False):
    a: bool | None
    b: bool | None
    c: bool | None
    d: bool | None
    e: bool | None
    f: bool | None
    g: bool | None
    h: bool | None
    i: bool | None
    j: bool | None
    k: bool | None
    l: bool | None
    m: bool | None
    n: bool | None
    o: bool | None
    p: bool | None


def foo() -> None:
    settings: GroupsSettingsDict = {}

    if "a" in settings:
        settings["a"]
    if "b" in settings:
        settings["b"]
    if "c" in settings:
        settings["c"]
    if "d" in settings:
        settings["d"]
    if "e" in settings:
        settings["e"]
    if "f" in settings:
        settings["f"]
    if "g" in settings:
        settings["g"]
    if "h" in settings:
        settings["h"]
    if "i" in settings:
        settings["i"]
    if "j" in settings:
        settings["j"]
    if "k" in settings:
        settings["k"]
    if "l" in settings:
        settings["l"]
    if "m" in settings:
        settings["m"]
    if "n" in settings:
        settings["n"]
    if "o" in settings:
        settings["o"]
    if "p" in settings:
        settings["p"]
```

Using 1.1.258: Completed in 0.467sec
Using 1.1.259: Completed in 177.808sec (each additional "'key' in settings" causes an exponentially slower evaluation as shown when running pyright --verbose --stats)

This is still valid in the latest release (1.1.262)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.263, which I just published.
